libs/log/default.go

```diff
@@ -0,0 +1,96 @@
+package log
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/rs/zerolog"
+)
+
+var _ Logger = (*DefaultLogger)(nil)
+
+// DefaultLogger implements a default logger that can be used within Tendermint
+// and that fulfills the Logger interface. The underlying logging provider is a
+// zerolog logger that supports typical log levels along with JSON and plain/text
+// log formats.
+//
+// Since zerolog supports typed structured logging and it is difficult to reflect
+// that in a generic interface, all logging methods accept a series of key/value
+// pair tuples, where the key must be a string.
+type DefaultLogger struct {
```

Comment:
I'd just make a constructor that takes a zerolog.Logger and returns one of our instances.

Reply:
We don't want folks having to even interact with or create zerolog. But we can have a constructor that accepts one for convenience.

Reply:
yeah, that makes sense eventually. 

keeping this type public, in effect exposes a zerolog implementation, so I think it's good to make this private regardless.